src/main/kotlin/cacophony/semantic/syntaxtree/SyntaxTree.kt

```diff
@@ -112,14 +112,34 @@ sealed class Definition(
                 areEquivalentTypes(type, other.type)
     }
 
-    class FunctionDeclaration(
+    sealed class FunctionDef(
         range: Pair<Location, Location>,
         identifier: String,
         val type: Type.Functional?,
-        val arguments: List<FunctionArgument>,
         val returnType: Type,
+    ) : Definition(range, identifier) {
+        override fun isEquivalent(other: Expression?): Boolean =
+            super.isEquivalent(other) &&
+                other is FunctionDeclaration &&
+                areEquivalentTypes(type, other.type) &&
+                areEquivalentTypes(returnType, other.returnType)
+    }
+
+    class ForeignFunctionDef(
```

Comment:
I see that we have a problem with naming and we swapped meanings of declaration and definition. Foreign functions can only be declared, but not defined. We could probably fix it and swap all declarations and definitions, if you don't want to do it here you may keep it as it is for integrity (and maybe order someone else to fix naming 😈).

Reply:
that was actually quite bad. I changed the naming, but this makes the diff much bigger

Reply:
but of course, it'll change snapshots once again :upside_down_face: